### ISSUE TITLE
kexec multiboot: unable to load modules 

### ISSUE THREAD

**Issue by jensdrenhaus** [u-root/u-root#1063]
trying to boot *fedora* via *tboot* ([tboot-GitHub](https://github.com/tklengyel/tboot)) using *kexec* fails: 
![tboot_boot_kexec](https://user-images.githubusercontent.com/14163031/50777540-30891880-129c-11e9-8d0d-b847d5fb0978.JPG)
Afterwards the machine reboots.

When booting the same *tboot* binary with the same kernel and initramfs as modules like this:
/boot/grub2/grub.cfg
``` 
menuentry ' Fodora with tboot'
      set root='hd0,msdos1'
      multiboot      /tboot.gz logging=serial,memory,vga
      module         /vmlinuz-4.18.16-300.fc29.x86_64 root=/dev/mapper/fedora-root ro # ... rest of cmdline
      module         /nitramfs-4.18.16-300.fc29.x86_64.img
```
then *tboot* output during startup looks like that and *fedora* starts successfully:
![tboot_boot_grub](https://user-images.githubusercontent.com/14163031/50778464-4ac3f600-129e-11e9-8d5b-dc51543e5afe.JPG)

It looks to me that the modules are not loaded properly so that *tboot* connot access them.

In *uroot* I use kexec as follows:
![kexec_load_cmd](https://user-images.githubusercontent.com/14163031/50778692-e6556680-129e-11e9-8df0-ca6899f9f4e8.JPG)
with the following output (don't be confused, I've added some extra output in the code, starteing with DEBUG: ...)
![kexec_load_debug](https://user-images.githubusercontent.com/14163031/50778834-33d1d380-129f-11e9-9207-b5d084014e38.JPG)

**Comment by n-canter** [comment]
I see one suspicious thing in your output: the modules' starts are not page aligned.
Multiboot spec doesn't require it but looks like many implementations do. I'll write a fix.

**Comment by n-canter** [comment]
Fixed by #1064
